src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -293,12 +293,11 @@ internal static void RunInternal<TState>(ExecutionContext executionContext, Cont
             edi?.Throw();
         }
 
-        internal static void RunFromThreadPoolDispatchLoop(ExecutionContext executionContext, ContextCallback callback, object state)
+        internal static void RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, object state)
```

Comment:
Could you please add an assert that the thread passed in is the same as the current thread?

Reply:
And similarly elsewhere